lib/analyzer/applications/java.ts

```diff
@@ -172,39 +169,39 @@ function unpackJarsTraverse({
       unpackedLevels < requiredLevelsOfUnpacking &&
       zipEntry.entryName.endsWith(".jar")
     ) {
-      isFatJar = true;
       const entryData = zipEntry.getData();
       const entryName = zipEntry.entryName;
-      jarPath = `${jarPath}/${entryName}`;
 
-      unpackJarsTraverse({
-        jarBuffer: entryData,
-        jarPath,
-        desiredLevelsOfUnpacking,
-        requiredLevelsOfUnpacking,
-        unpackedLevels,
-        jarBuffers,
-        coords,
-        dependencies,
+      nestedJars.push({
+        buffer: entryData as Buffer,
+        location: `${jarPath}/${entryName}`,
       });
     }
   }
 
-  if (!isFatJar) {
-    jarBuffers.push({
-      location: jarPath,
-      digest: jarBuffer,
-      coords,
-      dependencies,
-    });
-  }
-
-  return jarBuffers;
+  return {
+    location: jarPath,
+    buffer: jarBuffer,
+    coords,
+    dependencies,
+    nestedJars,
+  };
 }
 
+/**
+ * Manages the unpacking an array of JarBuffer objects and returns the resulting
+ * fingerprints. Recursion to required depth is handled here when the returned
+ * info from each JAR that is unpacked has nestedJars.
+ *
+ * @param {JarBuffer[]} jarBuffers
+ * @param {number} desiredLevelsOfUnpacking
+ * @param {number} unpackedLevels
```

Comment:
Curious about what this annotation means. Where is this loaded from? Some configuration file?

Reply:
Yes, we do apply a limit to the level of unpacking. By default, this is 1 level but controlled with the `--nested-jar-depth` flag to allow the customer control. We actually always unpack 1 level deeper than the `desiredLevelOfUnpacking` as that is how we find shaded JARs and coords for custom JARs. We only generate fingerprints for JARs found up to the `desiredLevelOfUnpacking` though.